pkg/daemons/agent/agent.go

```diff
@@ -199,16 +199,53 @@ func CheckCgroups() (kubeletRoot, runtimeRoot string, hasCFS, hasPIDs bool) {
 	// For Unified (v2) cgroups we can directly check to see what controllers are mounted
 	// under the unified hierarchy.
 	if cgroupsModeV2 {
```

Comment:
Rootless mode already does evacuation by itself: https://github.com/k3s-io/k3s/blob/245efe0d665838dfc228e3442efd11bcd3b39ec8/pkg/rootless/rootless.go#L150

So the evacuation log here should be omitted for rootless mode.
(Or remove the evacuation logic in `rootless.go`)

Reply:
@brandond what do you prefer here?

Reply:
@AkihiroSuda do you think it makes sense to consolidate the evaction stuff here? If it needs to be done regardless I think we might as well do it all in the agent package and simplify rootless a bit.

Reply:
SGTM

Reply:
So I'm a bit unsure here.. I feel like my code here could be replaced with `cgrouputil.EvacuateCgroup2("init")` (as it is in https://github.com/k3s-io/k3s/blob/245efe0d665838dfc228e3442efd11bcd3b39ec8/vendor/github.com/rootless-containers/rootlesskit/pkg/parent/cgrouputil/cgrouputil.go#L23), is that correct?
Or do we want to keep it like it is here and just remove this block here: https://github.com/k3s-io/k3s/blob/245efe0d665838dfc228e3442efd11bcd3b39ec8/pkg/rootless/rootless.go#L147-L153 ? :thinking:

Reply:
Yes, you can use cgrouputil.EvacuateCgroup2("init") here, for both rootful and rootless

Reply:
:heavy_check_mark: